admin/includes/init_includes/init_sanitize.php

```diff
@@ -237,9 +237,27 @@
 $sanitizer->addSimpleSanitization('PRODUCT_NAME_DEEP_REGEX', $group);
 
 $group = array('configuration_key', 'search', 'query_string');
+$sanitizer->addSimpleSanitization('STRICT_SANITIZE_VALUES', $group);
+
+$configs_with_special_characters = array(
+   'BREAD_CRUMBS_SEPARATOR',
+   'BEST_SELLERS_FILLER',
+   'CATEGORIES_SEPARATOR',
+   'CATEGORIES_SEPARATOR_SUBS',
+   'CATEGORIES_COUNT_PREFIX',
+   'CATEGORIES_SUBCATEGORIES_INDENT',
+   'EZPAGES_SEPARATOR_HEADER',
+   'EZPAGES_SEPARATOR_FOOTER',
+   'CURRENCIES_TRANSLATIONS',
+   'STOCK_MARK_PRODUCT_OUT_OF_STOCK',
+   'EMAIL_SMTPAUTH_PASSWORD',
+   'CATEGORIES_COUNT_SUFFIX',
+   'STORE_NAME_ADDRESS',
+   'PRODUCT_LIST_SORT_ORDER_ASCENDING',
+   'PRODUCT_LIST_SORT_ORDER_DESCENDING',
+); 
 
-$checks = new queryFactoryResult($db->link);
-// if current page is configuration, the configuration option does not have a val_function and saving the setting, then sanitize configuration_value to array.
+// if current page is configuration, handle separately
 if (defined('FILENAME_CONFIGURATION')
       && $_SERVER['SCRIPT_NAME'] == DIR_WS_ADMIN . (!strstr(FILENAME_CONFIGURATION, '.php') ? FILENAME_CONFIGURATION . '.php' : FILENAME_CONFIGURATION)

```

Comment:
Can this be simplified so that we only check for `!empty($_GET['cID'])` and pass page=configuration into the $group array?

ref comment at https://github.com/zencart/zencart/commit/ee3e8eb53344ac402fd4af378db441a5d7f37c84#r38459481

Reply:
Can we merge this now and improve it later?  I'm worried about breaking something by making a quick change.  This fixes a real bug in the system and should be taken.

Reply:
Other fixes have been proposed for v156. 
Are you planning to tell people "this" is the "new" fix for v156? 
If so, I'd like it to be more atomic than "add these three, in this specific order"

Reply:
This is not a fix for 1.5.6 - 1.5.6 needs email addresses to be handled as well.  Sorry, I could have sworn I said that this fix was targeted to 1.5.7 only in the description.  Fixing now.